src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -18,12 +18,14 @@ namespace Microsoft.DotNet.Tools.Install.Tool
 {
     internal class InstallToolCommand : CommandBase
     {
-        private readonly IToolPackageStore _toolPackageStore;
-        private readonly IToolPackageInstaller _toolPackageInstaller;
-        private readonly IShellShimRepository _shellShimRepository;
+        public delegate IShellShimRepository CreateShellShimRepository(DirectoryPath? nonGlobalLocation = null);
```

Comment:
@nguerrera use named delegate, am I doing it right? It does removed a lot of interfaces. It also has nonGlobalLocation in intellisense.

Reply:
That is fine. Usually I suggest avoiding public nested types though so I'd put this at namespace level.

Reply:
Moved to namespace level